src/driftpy/drift_user.py

```diff
@@ -69,17 +78,22 @@ def get_oracle_data_for_perp_market(
     ) -> Optional[OraclePriceData]:
         return self.drift_client.get_oracle_price_data_for_perp_market(market_index)
 
-    def get_perp_market_account(self, market_index: int) -> PerpMarketAccount:
+    def get_perp_market_account(self, market_index: int) -> Optional[PerpMarketAccount]:
         return self.drift_client.get_perp_market_account(market_index)
 
-    def get_spot_market_account(self, market_index: int) -> SpotMarketAccount:
+    def get_spot_market_account(self, market_index: int) -> Optional[SpotMarketAccount]:
         return self.drift_client.get_spot_market_account(market_index)
 
     def get_user_account_and_slot(self) -> DataAndSlot[UserAccount]:
-        return self.account_subscriber.get_user_account_and_slot()
+        if not self.subscribed:
+            raise SubscriptionException()
+        user_account = self.account_subscriber.get_user_account_and_slot()
+        if not user_account:
+            return self.initial_data
```

Comment:
feel like this would lead to extra confusion in long run where you end up w stale data